field_encrypt.module

```diff
@@ -117,33 +123,123 @@ function field_encrypt_form_field_add_form_builder($entity_type, \Drupal\field\E
-    //    }
-    //    elseif ($form_encryption === 0) {
-    //      $field_encrypt_process_entities->decryptStoredField($field_entity_type, $field_name);
-    //    }
+    // Only re-encrypt field data if this field contains actual data.
+    if ($field_storage_config->hasData()) {
+      // We need to process the field to either encrypt or decrypt the stored
+      // fields if the setting was changed.
+      $storage = $form_state->getStorage();
+      $storage['field_encrypt_update'] = TRUE;
+      $storage['field_encrypt_field_name'] = $field_storage_config->get('field_name');
+      $storage['field_encrypt_entity_type'] = $field_storage_config->get('entity_type');
+      $storage['original_encryption_settings'] = $original_encryption;
+      $form_state->setStorage($storage);
+    }
+  }
+}
+
+/**
+ * Submit handler to update encryption after saving field storage settings.
+ *
+ * @param $form
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ */
+function field_encrypt_update_encryption_form_submit(&$form, FormStateInterface $form_state) {
+  $storage = $form_state->getStorage();
+  $entity_type_manager = \Drupal::service('entity_type.manager');
+  if (isset($storage['field_encrypt_update']) && $storage['field_encrypt_update'] == TRUE) {
+    $entity_type_id = $storage['field_encrypt_entity_type'];
+    $entity_type = $entity_type_manager->getDefinition($entity_type_id);
+    $field_name = $storage['field_encrypt_field_name'];
+    $original_encryption_settings = $storage['original_encryption_settings'];
+
+    // Get entity IDs to update.
+    $entity_query = \Drupal::service('entity.query');
+    $query = $entity_query->get($entity_type_id);
+    // Check if the field is present.
+    $query->exists($field_name);
+    if ($entity_type->hasKey('revision')) {
+      $query->allRevisions();
+    }
+    $entity_ids = $query->execute();
+
+    $batch = array(
+      'title' => t('Updating field encryption'),
+      'operations' => array(
+        array('field_encrypt_update_field_encryption', array(array_keys($entity_ids), $field_name, $entity_type_id, $original_encryption_settings)),
+      ),
+      'finished' => 'field_encrypt_update_field_encryption_finished',
+    );
+    batch_set($batch);
+  }
+}
+
+/**
+ * Callback for batch updating of encryption fields.
+ *
+ * @param $entity_type
+ * @param $field_name
+ * @param $context
+ */
+function field_encrypt_update_field_encryption($entity_ids, $field_name, $field_entity_type, $original_encryption_settings, &$context) {
+//  $entity_type_manager = \Drupal::service('entity_type.manager');
+  $process_entities = \Drupal::service('field_encrypt.process_entities');
+  if (empty($context['sandbox'])) {
+    $context['sandbox']['entity_ids'] = $entity_ids;
+    $context['sandbox']['progress'] = 0;
+    $context['sandbox']['max'] = count($entity_ids);
+  }
+
+  // Process entities by groups of 5.
+  $count = min(5, count($context['sandbox']['entity_ids']));
```

Comment:
Do you think we can/should make this configurable? Maybe in the settings form, we allow the user to select how many get processed at once? Default it to 5?